lasy/profiles/transverse/gaussian_profile.py

```diff
@@ -19,11 +19,28 @@ class GaussianTransverseProfile(TransverseProfile):
     ----------
     w0 : float (in meter)
         The waist of the laser pulse, i.e. :math:`w_0` in the above formula.
+
+    wavelength : float (in meter)
+        The main laser wavelength :math:`\\lambda_0` of the laser.
+
+    z_init : float (in meter), optional
+        Position at which the pulse is initialized, relative to the position
+        of the focal plane. (The focal plane is at `z=0`.)
+
+    .. warning::
+
+        In order to initialize the pulse out of focus, you can either:
+            - Use a non-zero `z_init`
+            - Use `z_init=0` and then call `laser.propagate(z_init)`
+        Both methods are in principle equivalent, but note that the first
+        method uses the paraxial approximation, while the second method does
+        not make this approximation.
     """
 
-    def __init__(self, w0):
+    def __init__(self, w0, wavelength, z_init=0):
```

Comment:
Could we make `wavelength` an optional parameter (mandatory if z_init is not 0`)? None of the other transverse profile require the wavelength.